### PR TITLE
fix(backend): surface analysis artifact feedback

### PR DESCRIPTION
**Pull request by qinsz01**
## Summary

Describe the problem and fix in 2-5 bullets:

* Problem: failed YJK analysis artifacts could return `success:false`, but the LangGraph `run_analysis` tool message still summarized the run as `success:true` with almost no diagnostic detail.
* Why it matters: the agent could not read artifact feedback such as YJK driver stderr/stdout logs, so it could not explain generated YJK errors or guide the user toward the failing phase.
* What changed: failed analysis responses now surface `success:false`, error code, message, and selected diagnostics to the model; YJK runtime errors now include work directory, driver artifact paths, recent steps, and capped stdout/stderr tails; the Python analysis API preserves structured exception metadata.
* What did NOT change (scope boundary): no UI, schema, storage, configuration, or successful-analysis result format changes.

## Change Type (select all)

* [x] Bug fix
* [ ] Feature
* [ ] Refactor required for the fix
* [ ] Docs
* [ ] Security hardening
* [ ] Chore/infra

## Scope (select all touched areas)

* [x] Gateway / orchestration
* [x] Skills / tool execution
* [ ] Auth / tokens
* [ ] Memory / storage
* [x] Integrations
* [x] API / contracts
* [ ] UI / DX
* [ ] CI/CD / infra

## Linked Issue/PR

* Closes N/A
* Related N/A
* This PR fixes a bug or regression

## Root Cause / Regression History (if applicable)

* Root cause: `run_analysis` treated any returned analysis artifact as a successful tool execution summary, even when the artifact payload itself had `success:false`.
* Missing detection / guardrail: there was no regression test asserting that failed analysis artifacts are visible to the model as failures.
* Prior context (`git blame`, prior PR, issue, or refactor if known): Unknown.
* Why this regressed now: YJK failures are delivered through the analysis response artifact rather than thrown as a TypeScript-side tool exception, so the compact tool summary hid the failure details.
* If unknown, what was ruled out: the full artifact was still stored in state; the missing piece was the compact ToolMessage content consumed by the model.

## Regression Test Plan (if applicable)

For bug fixes or regressions, name the smallest reliable test coverage that should have caught this. Otherwise write `N/A`.

* Coverage level that should have caught this:
  * Unit test
* Target test or file: `backend/src/agent-langgraph/__tests__/analysis-tool-summary.test.mjs`
* Scenario the test should lock in: a `success:false` YJK analysis artifact produces a `success:false` tool summary containing error code, message, and diagnostics such as stderr tail.
* Why this is the smallest reliable guardrail: the bug was in summary construction before the model sees the tool result, so a focused unit test catches it without requiring a live YJK installation.
* Existing test that already covers this (if any): None.
* If no new test is added, why not: N/A.

## User-visible / Behavior Changes

When YJK analysis fails, the assistant should now be able to explain the failing phase and relevant driver log tail instead of acting as if analysis succeeded or giving a generic artifact-reading failure. No settings or defaults changed.

## Diagram (if applicable)

    Before:
    YJK failure artifact -> run_analysis ToolMessage { success: true } -> model misses logs

    After:
    YJK failure artifact -> run_analysis ToolMessage { success: false, diagnostics } -> model can explain logs

## Security Impact (required)

* New permissions/capabilities? (`Yes/No`): No
* Secrets/tokens handling changed? (`Yes/No`): No
* New/changed network calls? (`Yes/No`): No
* Command/tool execution surface changed? (`Yes/No`): No
* Data access scope changed? (`Yes/No`): Yes
* If any `Yes`, explain risk + mitigation: the model now receives capped tails and paths from artifacts produced by the current YJK run. The runtime does not read additional arbitrary files, and stdout/stderr snippets are capped before being added to metadata/tool summaries.

### Environment

* OS: Windows
* Runtime/container: backend Node/Jest/TypeScript; Python 3.12 venv for syntax checks
* Model/provider: N/A
* Integration/channel (if any): LangGraph `run_analysis`, YJK static analysis runtime
* Relevant config (redacted): N/A

### Steps

1. Trigger or mock a failed YJK analysis response with `success:false` and stderr artifact feedback.
2. Build the `run_analysis` tool summary.
3. Inspect the returned ToolMessage summary and diagnostics.

### Expected

* The tool summary reports `success:false` and includes the error code, message, YJK artifact paths, recent steps, and capped stdout/stderr tails where available.

### Actual

* Before this PR, the tool summary reported `success:true` and omitted the YJK diagnostic context even when the artifact payload was a failure.

## Evidence

* Passing focused regression: `npm test --prefix backend -- --runInBand src/agent-langgraph/__tests__/analysis-tool-summary.test.mjs`
* Passing LangGraph tool suite: `npm test --prefix backend -- --runInBand src/agent-langgraph/__tests__`
* Passing lint: `npm run lint --prefix backend`
* Passing Python syntax check: `$HOME\.structureclaw\.venv\Scripts\python.exe -m py_compile backend/src/agent-skills/analysis/runtime/api.py backend/src/agent-skills/analysis/yjk-static/runtime.py`

## Human Verification (required)

What you personally verified (not just CI), and how:

* Verified scenarios: failed analysis artifacts now produce failed tool summaries with diagnostic fields; YJK runtime failure branches include artifact feedback; existing LangGraph tool tests still pass.
* Edge cases checked: timeout, empty stdout, invalid JSON, and driver `status:error` branches now use the same artifact-feedback formatter.
* What you did not verify: a live licensed YJK GUI/API run was not executed in this session.

## Review Conversations

* I replied to or resolved every bot review conversation I addressed in this PR.
* I left unresolved only the conversations that still need reviewer or maintainer judgment.

No review conversations exist yet.

## Compatibility / Migration

* Backward compatible? (`Yes/No`): Yes
* Config/env changes? (`Yes/No`): No
* Migration needed? (`Yes/No`): No
* If yes, exact upgrade steps: N/A

## Risks and Mitigations

* Risk: YJK stderr/stdout can be noisy.
  * Mitigation: log text is capped before being added to exception metadata and the model-facing summary.
* Risk: local artifact paths are surfaced in model context.
  * Mitigation: only paths from the current generated analysis work directory are included; no arbitrary extra files are read.

### CODE REVIEW

File: backend/src/agent-langgraph/tools.ts [hunk 1]
```diff
@@ -60,6 +60,78 @@ function toolResult(
...
+  const success = result.success === false || status === 'error' ? false : true;
+
+  if (!success) {
+    const errorCode = result.error_code || result.errorCode || 'ANALYSIS_EXECUTION_FAILED';
+    const diagnostics = pickAnalysisDiagnostics(result, meta);
+    return {
+      success: false,
+      skillId: args.skillId,
+      errorCode,
+      message: compactText(result.message) || 'Analysis execution failed',
```
[the remark this answers is not in the document]
Comment (qinsz01) [reply backend/src/agent-langgraph/tools.ts:150]: Addressed in a5b00b9: errorCode is now normalized to the first non-empty string value, otherwise falling back to ANALYSIS_EXECUTION_FAILED.

File: backend/src/agent-langgraph/tools.ts [hunk 2]
```diff
@@ -60,6 +60,78 @@ function toolResult(
...
+  return typeof value === 'object' && value !== null && !Array.isArray(value);
+}
+
+function compactText(value: unknown, limit = ANALYSIS_MESSAGE_LIMIT): string | undefined {
+  if (typeof value !== 'string') return undefined;
+  const text = value.trim();
+  if (!text) return undefined;
+  if (text.length <= limit) return text;
+  return `${text.slice(0, limit)}\n...[truncated ${text.length - limit} chars]`;
+}
```
[the remark this answers is not in the document]
Comment (qinsz01) [reply backend/src/agent-langgraph/tools.ts]: Addressed in a5b00b9: analysis failure message compaction now keeps both the beginning and the recent tail so appended YJK stderr/stdout diagnostics survive truncation.

File: backend/src/agent-langgraph/tools.ts [hunk 3]
```diff
@@ -60,6 +60,78 @@ function toolResult(
...
+
+function pickAnalysisDiagnostics(
+  result: Record<string, unknown>,
+  meta: Record<string, unknown>,
+): Record<string, unknown> | undefined {
+  const diagnostics: Record<string, unknown> = {};
+  const copy = (targetKey: string, value: unknown) => {
+    if (value === undefined || value === null || value === '') return;
+    diagnostics[targetKey] = typeof value === 'string' ? compactText(value, 2000) ?? value : value;
+  };
```
[the remark this answers is not in the document]
Comment (qinsz01) [reply backend/src/agent-langgraph/tools.ts:112]: Addressed in a5b00b9: Tail diagnostics now use tail-preserving compaction, and the regression test checks that the latest stderr marker is retained.

File: backend/src/agent-skills/analysis/yjk-static/runtime.py [hunk 4]
```diff
@@ -124,6 +127,98 @@ def _read_json(path: Path) -> dict | None:
...
+
+    meta: Dict[str, Any] = {label: str(path) for label, path in paths.items()}
+    if stdout_tail:
+        meta["stdoutTail"] = stdout_tail
+    if stderr_tail:
+        meta["stderrTail"] = stderr_tail
+    if steps_tail:
+        meta["stepsTail"] = steps_tail
+    if detail:
+        meta["yjkErrorDetail"] = detail
```
[the remark this answers is not in the document]
Comment (qinsz01) [reply backend/src/agent-skills/analysis/yjk-static/runtime.py]: Addressed in a5b00b9: YJK detail payloads are now recursively sanitized/capped, and the human-readable message only includes an allowlisted detail summary.

File: backend/src/agent-langgraph/tools.ts [hunk 5]
```diff
@@ -60,6 +60,78 @@ function toolResult(
...
+
+export function buildAnalysisToolSummary(args: {
+  result: unknown;
+  skillId?: string;
+}): Record<string, unknown> {
+  const result = isRecord(args.result) ? args.result : {};
+  const meta = isRecord(result.meta) ? result.meta : {};
+  const data = isRecord(result.data) ? result.data : undefined;
+  const status = typeof result.status === 'string' ? result.status : undefined;
+  const success = result.success === false || status === 'error' ? false : true;
```
[the remark this answers is not in the document]
Comment (qinsz01) [reply backend/src/agent-langgraph/tools.ts]: Addressed in a5b00b9: simplified the success expression to result.success !== false && status !== "error".

**Review by guyi2000 — APPROVED** [review]
